quickfixj-core/src/main/java/quickfix/mina/initiator/ConnectionException.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) quickfixengine.org  All rights reserved.
+ *
+ * This file is part of the QuickFIX FIX Engine
+ *
+ * This file may be distributed under the terms of the quickfixengine.org
+ * license as defined by quickfixengine.org and appearing in the file
+ * LICENSE included in the packaging of this file.
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING
+ * THE WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE.
+ *
+ * See http://www.quickfixengine.org/LICENSE for licensing information.
+ *
+ * Contact ask@quickfixengine.org if any conditions of this licensing
+ * are not clear to you.
+ ******************************************************************************/
+
+package quickfix.mina.initiator;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+
+public class ConnectionException extends IOException {
```

Comment:
Sorry for the delay. I have one minor remark: could we rename the class to `ConnectException` so that it matches the name of the callback `onConnectException`. Thank you.

Reply:
There's [`java.net.ConnectException`](https://docs.oracle.com/javase/8/docs/api/java/net/ConnectException.html), but I guess that is fine as classes in most applications wouldn't have both this and that in the namespace anyway.

Reply:
Thanks, that's a valid point. It even also extends `IOException`. But as you said it should be OK.